main.go

```diff
@@ -161,6 +186,53 @@ func main() {
 	}
 }
 
+func getRulesForTenants(ctx context.Context, observatoriumURL url.URL, tenants []tenant) (io.Reader, error) {
+	queue := make(chan tenant)
+	respc := make(chan []byte)
+
+	go func() {
+		var wg sync.WaitGroup
+		wg.Add(len(tenants))
+
+		for i := 0; i < fetchConcurrency; i++ {
+			go func() {
+				for tenant := range queue {
+					observatoriumURL.Path = path.Join("/api/metrics/v1", tenant.Name, "/api/v1/rules/raw")
+					// TODO(onprem): Handle errors.
```

Comment:
This seems actually pretty crucial. Getting the rules is the core functionality of this whole project; if this fails, we need to know and retry. I think we should address this :+1: